---
title: "JJTOOLS Package document"
author: "José Jordán"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{jjtools document}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r message=FALSE, warning=FALSE}
library(jjtools)
library(dplyr)
library(readr)
library(data.table)
```


## Vignette Info

Sometimes it is necessary to process a data set to add information of interest or to calculate some value or useful statistic. This simple package allows you to process RNASeq count data obtained by RSubread, which processes BAM files containing the reads aligned with a reference genome. This package is also used for any dataset that contains, along with raw counts, a column called "Gene ID", with the Gene ID identifier in NCBI (Entrez) format. If you need to modify your Gene ID data, you can use this tool (available online in Ensembl):

[https://www.ensembl.org/biomart/martview/18870173edad33c219d9977902fa95fe](https://www.ensembl.org/biomart/martview/18870173edad33c219d9977902fa95fe)


## Full exemple step by step

Step-by-step analysis of an RNASeq dataset with the jjtools package:

### Step 1: add info

```{r}
addinfo(data_RNASeq_raw_counts,hg38) #Selected hg38_RefSeq_exon dataset (in Rsubread folder)
```

You can add an addicional info with the jjtools command *extract()*. This function allows to extract from, for example hg38_RefSeq_exon dataset, the length of the gene (in bp and kb). 

<style>
p.comment {
background-color: #f94545;
color:white;
padding: 0px;
border: 1px solid black;
margin-left: 0px;
text-align: center;
border-radius: 1px;
font-style: italic;
}
</style>

<p class="comment">
_You must transform to character both columns and after join both tables_
</p>


```{r}
# You can use the function extract() to obtain gene data from Rsubread package and obtain the gene_lenght_file"
datastart <- data.frame("GeneID"=hg38$GeneID,"Start"=hg38$Start)
dataend <- data.frame("GeneID"=hg38$GeneID,"End"=hg38$End)
setDT(datastart)[ , .SD[which.min(Start)], by = "GeneID"]  -> min_start
setDT(dataend)[ , .SD[which.max(End)], by = "GeneID"]  -> max_start

left_join(max_start,min_start, by="GeneID")->joined_max_min_gene

data.frame("GeneID"=joined_max_min_gene$GeneID, "Gene_lenght"=joined_max_min_gene$End-joined_max_min_gene$Start) -> gene_lentgh
gene_lentgh$Gene_lenght/1000 -> gene_lentgh$kb_Gene_lenght
gene_lentgh -> gene_lenght_file

transform(gene_lenght_file, Gene_lenght = as.character(Gene_lenght),
          kb_Gene_lenght = as.character(kb_Gene_lenght))->gene_lenght_file2
left_join(raw_counts_addinfo,gene_lenght_file2, 
          by="GeneID")->raw_counts_addinfo2
raw_counts_addinfo2[,c(1:6,11:12)] #See that the last two columns are 'chr'
```

### Step 2: counts normalization (pkm or tpm method)

We calculate the pkm or tpm from our raw counts; in this example we will use tpm and we will use raw_counts_addinfo object. For more info about pkm and tpm, see help functions ?pkm_method() and ?tpm_method().

```{r}
tpm_method(raw_counts_addinfo,gene_lenght_file)
# Testing that the sum of each column is = 1000000, for example with column S1:
sum(tpm_sample$S1)
```

You can calculate some statistical values to see the average value of each gene between the different samples with the calc() function:^[If you select some rows, the function will be show only these genes. For example writing calc(tpm_sample[1:8,])]

```{r}
calc(tpm_sample)
```

```{r}
norm_counts_added_stats[,c(1:3,7,11:13)]
```

### Step 3: filter by genes of interest (surfaceome) 

You can filter by surfaceome genes; in jjtools there is a dataset (\link{surfy}) with 2911 genes ready to use with the surface() function 

```{r}
surface(norm_counts_added_stats) #We use the result in the previous chunk -calc()-
counts_filtered_by_surfaceome
```

### Step 4a: Find out the most informatives data in your dataset

With inform() you can find out the most informatives in your dataset. In this case we will use tpm_sample object obtained in Step 2:

<p class="comment">
_Remember that you can obviate the value of filter; by default is configured to be = 1_
</p>

```{r}
# First of all you MUST use only the matrix with the dataset, so you must remove columns without data:
tpm_sample[,2:6] -> new_sample
# Use a cut-off value to filter the dataset. In this case we will use 10000:
filter_value=10000
# Apply the function:
inform(new_sample,filter_value)
```

##### Step 4b: Find out the group of most informatives data in your dataset

Now we will compute something similar to the previous section (4a). In this case we will obtain the group of columns (cells) that maximizes unique genes expressed. We will use an online dataset for our comodity:

```{r}
data_cells <- read_table2("https://www.ebi.ac.uk/gxa/experiments-content/E-PROT-25/resources/ExperimentDownloadSupplier.Proteomics/tsv", 
    col_names = FALSE, skip = 5)

```

First of all, we processed the file. We remove the NAs and filter the number of genes by surfaceome:

```{r}
replace(data_cells, is.na(data_cells), 0) -> file_line_cells_NA
WantedData=file_line_cells_NA[file_line_cells_NA$X2 %in% surfy$`Gene symbol UniProt`, ]
WantedData
#Now you can apply tpm_method() function in jjtools or other. Or use simply these values, as in this example to ilustrate the group() function in jjtools
```

Now we will specify in _group()_ our matrix, the number of simulations (1000), the cut-off value to filter the matrix data to binary (250), the number of cells to select (10), the set.seed value (337 for example) and if the 1st column is when start our matrix data (in this cause must be 3 because we have 2 columns with info -gene names and code-) 

```{r}
group(WantedData,1000,250,10,337,3)
names_group_cells
```

