#' @import readr
#' @import dplyr
#' @import data.table
#' @import mygene
#' @title Add some information
#' @description It adds extra information to raw_counts generated by RSubread and also eliminates those GeneID whose value in all samples is = 0.
#' @details The result is returned in a R object  (raw_counts_addinfo).
#' @param raw_counts Object in R that contains the table of counts generated by Rsubread.
#' @param gene_data Object in R that contains gene info (see \link{hg38} dataset as example!).
#' @examples
#' # We prepare the sample data:
#' c(1,2,10,55,100) -> genes
#' c(0,5,10,20,50) -> sampleA
#' c(0,0,15,30,50) -> sampleB
#' c(0,1,2,3,100) -> sampleC
#' data.frame("GeneID"= genes,"A_counts"= sampleA,"B_counts"=sampleB,"C_counts"=sampleC) -> test_counts
#' str(test_counts)
#' # We apply the function:
#' addinfo(test_counts)
#' # Select the RefSeq file from the Rsubread package.
#' # WAIT...
#' raw_counts_addinfo # R object where it is saved
#' # You can see that the "GeneID" = 1 is missing because all counts in samples are 0.
#' ####################################################################
#' addinfo(data_RNASeq_raw_counts) # Internal dataset
#' # You can see that the "GeneID" = 1,4 & 3000 is missing because all counts in samples are 0.
#' @author Jose Jordan
#' @export
addinfo <- function(raw_counts,gene_data) {
  message('hg38 RefSeq file from the Rsubread package will be use as default to extract the
chromosome info and strand in which the GeneID identifier appears')
  datta_hg38 <- data.frame(GeneID = c(1, 2, 3, 4,10,15,20,50,55,100,1000,2000,3000,3050,100507050),
                           Chr = c("chr19", "chr12", "chr12", "chr?", "chr8", "chr17","chr9","chr22", "chr3","chr20","chr18","chrX","chr17","chr16","chr11"), Start=c(58346806,9067708,9228533,0,18391245,76452853,137007234,41469125,132317367,44619522,27932879,130064920,8002670,142053,75240905), End=c(58347029,9068224,9228698,0,18391345,76452931,137007964,41469182,132317576,44619847,27933124,130067525,8002734,143089,75243704),Strand=c("-","-","-","+","+","+","-","+","+","-","-","-","+","+","+")
  )
  if (missing(gene_data)) {
    datta_hg38->gene_data
    message('
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
!!++++++++++++++++++++++++++++++++++++++++++!!
!!+   Partial hg38 data used as default.   +!!
!!+     See hg38 function in jjtools       +!!
!!++++++++++++++++++++++++++++++++++++++++++!!
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
            ')
  }
  as.data.frame(gene_data)->hg38_data
  print(head(hg38_data))
  #myVector <- c('GeneID', 'Chr', 'Strand')
  #hg38_data[, myVector]->geneid_chr
  #arrange(hg38_data, GeneID, Chr,Strand)->geneid_chr  FUNCIONAN pero en check no va
  hg38_data %>% dplyr::select(GeneID,Chr,Strand) -> geneid_chr
  unique(geneid_chr) -> geneid_chr_unique
  names(geneid_chr_unique)[1] <- "GeneID"
  names(geneid_chr_unique)[2] <- "Chr"
  names(geneid_chr_unique)[3] <- "Strand"

  # message('Selecciona tu fichero raw_counts obtenido con Rsubread. Recuerda que la columna primera,
# la que contiene el identificador de cada gen, deberÃ­a llamarse "GeneID"')
  # read_delim(file.choose(), ";", escape_double = FALSE, trim_ws = TRUE) -> raw_counts
  raw_counts$GeneID <- as.numeric(raw_counts$GeneID)
  left_join(raw_counts,geneid_chr_unique, by="GeneID") -> raw_counts_join
  raw_counts_join %>% relocate(c("Chr","Strand"), .after = "GeneID") -> raw_counts_join2
  raw_counts_join2[apply(raw_counts_join2[,-c(1:3)], 1,function(x) !all(x==0)),] -> raw_counts_add
  raw_counts_add[raw_counts_add$GeneID %in% names(which(table(raw_counts_add$GeneID) < 2)), ] -> raw_counts_add2

  raw_counts_add2$GeneID -> list
  data.frame(list) -> listigenes
  getGenes(listigenes[,]) -> genes_listado
  genes_listado$entrezgene -> aa
  genes_listado$name -> bb
  genes_listado$symbol -> cc
  data.frame(aa,bb,cc) -> data_frame_genes_listado
  c("GeneID","Name.Gene","Symbol") -> names
  names(data_frame_genes_listado) <- names
  transform(data_frame_genes_listado, "GeneID" = as.numeric(GeneID)) -> data_frame_genes_listado2
  left_join(raw_counts_add2,data_frame_genes_listado2, by = "GeneID") -> raw_counts_add3
  raw_counts_add3 %>% relocate(c("Name.Gene","Symbol"), .after = "GeneID") -> raw_counts_add4
  names(raw_counts_add4)[2]<-"Gene name"
  raw_counts_add4 ->> raw_counts_addinfo
  message('FINAL TABLE GENERATED:')
  print(raw_counts_addinfo)
  nrow(raw_counts)-nrow(raw_counts_addinfo) -> rows_rmv
  message('Final table generated with your raw_counts + extra information')
  message('# of deleted rows whose counts are in ALL data columns = 0:')
  print(rows_rmv)
  message('In the **raw_counts_addinfo** object you have the table available')
  message('TIP: to save on PC use the following command in R ("utils" package):')
  message('write.table(raw_counts_addinfo, file="raw_counts_addinfo.csv", sep =";")')



  rm(raw_counts_add2,gene_data,datta_hg38,data_frame_genes_listado,cc,bb,genes_listado,aa,raw_counts_add4,listigenes,raw_counts_add3,list,names,data_frame_genes_listado2,raw_counts,geneid_chr,geneid_chr_unique,raw_counts_join,raw_counts_join2,raw_counts_add,rows_rmv)
}
